### PR TITLE
[FIX]: Users demo

### LINKED ISSUE #43: no function clause matching in DemoWeb.UserLive.Edit.mount/2
Hi,

Here is the error encountered when I try to edit or show a user.

`** (FunctionClauseError) no function clause matching in DemoWeb.UserLive.Show.mount/2
        (demo) lib/demo_web/live/user_live/show.ex:23: DemoWeb.UserLive.Show.mount(%{}, %Phoenix.LiveView.Socket{assigns: %{}, changed: %{}, connected?: false, endpoint: DemoWeb.Endpoint, fingerprints: {nil, %{}}, id: "phx-RlPCg8nt", parent_pid: nil, private: %{assigned_new: {%{}, []}, connect_params: %{}}, redirected: nil, root_pid: nil, router: DemoWeb.Router, temporary: %{}, view: DemoWeb.UserLive.Show})
        (phoenix_live_view) lib/phoenix_live_view/view.ex:467: Phoenix.LiveView.View.call_mount!/3
        (phoenix_live_view) lib/phoenix_live_view/view.ex:435: Phoenix.LiveView.View.call_mount_and_handle_params!/5
        (phoenix_live_view) lib/phoenix_live_view/view.ex:300: Phoenix.LiveView.View.static_render/3
        (phoenix_live_view) lib/phoenix_live_view/controller.ex:37: Phoenix.LiveView.Controller.live_render/3
        (phoenix) lib/phoenix/router.ex:288: Phoenix.Router.__call__/2
        (demo) lib/demo_web/endpoint.ex:1: DemoWeb.Endpoint.plug_builder_call/2
        (demo) lib/plug/debugger.ex:122: DemoWeb.Endpoint."call (overridable 3)"/2
        (demo) lib/demo_web/endpoint.ex:1: DemoWeb.Endpoint.call/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:42: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /Users/charles/workspace/phoenix/phoenix_live_view_example/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /Users/charles/workspace/phoenix/phoenix_live_view_example/deps/cowboy/src/cowboy_stream_h.erl:296: :cowboy_stream_h.execute/3
        (cowboy) /Users/charles/workspace/phoenix/phoenix_live_view_example/deps/cowboy/src/cowboy_stream_h.erl:274: :cowboy_stream_h.request_process/3
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3`

I tried multiple elixir versions (in 1.8.2 now).

### PR DESCRIPTION
**Pull request by snewcomer**
close #43 #51

### CODE REVIEW

**Comment by jg-made** [comment]
@snewcomer I might be doing something wrong locally but when I run your branch and try to delete a user I get an error message:
```value `%{"altKey" => false, "ctrlKey" => false, "metaKey" => false, "pageX" => 1255, "pageY" => 180, "screenX" => 1255, "screenY" => 309, "shiftKey" => false, "x" => 1255
, "y" => 180}` in `where` cannot be cast to type :id in query:```
It looks like the `link` being used to delete a user in the index.html template is not sending the user id but instead sending details about the click event.

**Comment by snewcomer** [comment]
@jg-made great find 👍  I fixed and added a test

**Comment by jg-made** [comment]
@snewcomer thanks again! Great work on making it compatible with modern live view.

**Comment by jg-made** [comment]
@snewcomer I have another question please. I see that in `user_live/index.ex` we have a `handle_info/2` function which looks like it is written to handle all kinds of `:user` changes (`:updated`, `:deleted`, etc.).
However, if I have two separate pages open, say the first one on the `index` list view and say in the second page I delete or update or add a user, the list on the first page does not update. I think maybe the index view has not subscribed to the user socket properly?

File: lib/demo_web/live/user_live/index.ex [hunk 1]
```diff
@@ -11,7 +11,7 @@ defmodule DemoWeb.UserLive.Index do
     {:ok, assign(socket, page: 1, per_page: 5)}
   end
 
-  def handle_params(params, url, socket) do
+  def handle_params(params, _url, socket) do
```
Comment (jg-made) [inline lib/demo_web/live/user_live/index.ex]: add this line
```
if connected?(socket), do: Demo.Accounts.subscribe()
```

Comment (snewcomer) [reply lib/demo_web/live/user_live/index.ex]: Thank you thank you!  Updated and nice job 👍

**Comment by snewcomer** [comment]
too many merge conflicts

**Pull request closed without merging**